examples/shiftbot/src/map.ts

```diff
@@ -0,0 +1,227 @@
+        if (map_without_doors.foreground[1][y].name == 'key_door_green') {
+            map_without_doors.foreground[0][y] = new Block('gravity_right')
+            map_without_doors.foreground[1][y] = new Block('gravity_right')
+            map_without_doors.foreground[2][y] = new Block('gravity_right')
+            map_without_doors.foreground[3][y] = new Block('gravity_right')
+            map_without_doors.foreground[0][y + 1] = new Block('hazard_stripes')
+        }
+    }
+}
+
+function construct_map() {
+    const map_no_border = get_map()
+    console.log(map_no_border.meta) // TODO keep till bug gets fixed
+    map.paste(1, 1, map_no_border)
+    map_without_doors.paste(1, 1, map_no_border)
+    encode_doors()
+    return map_no_border.meta
+}
+
+function find_map(search_string: string): [string, string] | null {
+    let maps: [string, { name: string }][] = fs.readdirSync(MAPS_PATH)
+        .filter(p => p.startsWith('map'))
+        .map(name => [name, Structure.metaFromString(
+            fs.readFileSync(path.join(MAPS_PATH, name))
+                .toString('ascii'))])
+    const map = maps.find(p => p[1].name == search_string)
+    if (map) return [map[0], map[1].name]
+    maps = maps.filter(p => p[1].name.includes(search_string))
+    if (maps.length == 0) return null
+    return [maps[0][0], maps[0][1].name]
+}
+
+export async function open_door() {
+    await client.block(TOP_LEFT.x + 22, TOP_LEFT.y + map.height - 2, 1, 'gravity_right')
+    await client.block(TOP_LEFT.x + 26, TOP_LEFT.y + map.height - 2, 1, 'gravity_right')
+    await client.block(TOP_LEFT.x + 28, TOP_LEFT.y + map.height - 2, 1, 'gravity_right')
+    return client.block(TOP_LEFT.x + 29, TOP_LEFT.y + map.height - 2, 1, 'gravity_right')
+}
+
+export async function create_win_zone() {
+    // Close Map Door
+    await client.world?.paste(TOP_LEFT.x, TOP_LEFT.y, map, { animation: Animation.RANDOM, write_empty: true })
+    // Create Win Zone
+    await client.block(TOP_LEFT.x + 26, TOP_LEFT.y + map.height - 2, 1, 'crown')
+    await client.block(TOP_LEFT.x + 28, TOP_LEFT.y + map.height - 2, 1, 0)
+    return client.block(TOP_LEFT.x + 29, TOP_LEFT.y + map.height - 2, 1, 'hazard_stripes')
+}
+
+export function set_spawn() {
+    return client.block(TOP_LEFT.x + 28, TOP_LEFT.y + map.height - 2, 1, 'spawn_point')
+}
+
+export function remove_spawn() {
+    return client.block(TOP_LEFT.x + 28, TOP_LEFT.y + map.height - 2, 1, 0)
+}
+
+export function close_door() {
+    return client.block(TOP_LEFT.x + 22, TOP_LEFT.y + map.height - 2, 1, 'hazard_stripes')
+}
+
+export async function build_map() {
+    const data = fs.readFileSync(path.join(MAPS_PATH, 'filled.yaml')).toString()
+    const structure = Structure.fromString(data)
+    map.paste(0, 0, structure)
+    map_without_doors.paste(0, 0, structure)
+    const meta = construct_map()
+    await client.world?.paste(TOP_LEFT.x, TOP_LEFT.y, map_without_doors, { animation: Animation.RANDOM, write_empty: true })
```

Comment:
Last parameter could be taken by a factory: `Animations.Randomized(true)`
The factory can then offer more animation options

Reply:
@LeoGiaco Animation Options can be overriden. It's a generator function that yields the order from an array of placement data. The default is `FIFO`, bombot uses top left to right bottom `L2RT2B`.